cocos/core/data/object.ts

```diff
@@ -620,7 +620,7 @@ declare namespace CCObject {
  * @return @en Whether it is a CCObject boolean value. @zh 是否为CCObject的布尔值。
  * @engineInternal
  */
-export function isCCObject (object: any): boolean {
+export function isCCObject (object: any): object is CCObject {
```

Comment:
I think `boolean` is correct.

Reply:
we need this type assertion to tell TS language service that if this method returns true, `object` should be an instance of `CCObject`

it's useful when we need to do some **Type Narrowing**, for example

```ts
function test (obj: unknown) {
  if (isCCObject (obj)) {
      obj.isValid   // we can access this property defined in `isCCObject ` class
  } else {
    obj.isValid  // this should emit an error
  }
}
```

Reply:
Got it, it is wired.